kolibri/plugins/user/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.deprecation import MiddlewareMixin
+
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import KolibriAnonymousUser
+
+ALLOWED_PATH_LIST = [
+    'deviceprovision',
+    'kolibri:setupwizardplugin:setupwizard',
+    'kolibri:set_language',
+    'session-list',
+    'admin:index',
+]
+
+
+class RedirectToSignInPageIfNoGuestAccessAndNoActiveSession(MiddlewareMixin):
+    """
+    Redirect to the sign in page if facility does not allow guest access and user
+    has not signed into the facility.
+    """
+
+    def process_request(self, request):
+        # Don't redirect for URLs that are required for the setup wizard
+        allowed_paths = [reverse(name) for name in ALLOWED_PATH_LIST]
+        if settings.DEBUG:  # allow anonymous rest api browsing
+            allowed_paths.append('/api')
+        if any(request.path.startswith(path_prefix) for path_prefix in allowed_paths):
+            return
+
+        if isinstance(request.user, KolibriAnonymousUser):
+            dataset = getattr(Facility.get_default_facility(), 'dataset', None)
```

Comment:
Don't want to change the design, I just think we should note this in the user facing design, as setting the facility setting will *only* make a difference if they are doing this for the default facility (so perhaps we should just hide this setting if someone happens to not be modifying the default facility for this device).

Reply:
regarding this, could we perhaps change the implementation of `get_default_facility` to read from a cookie in that way an admin can change the setting for any facility they happen to be logged into on a device?

Reply:
The default facility is the default facility!

I think it is better just to be explicit that the default facility has this kind of 'override' behaviour for the whole device when we let them set the setting. (the other alternative is that this is actually a device setting, but I think that might be even more confusing for users)